### PR TITLE
added endgame integration

### PR DESCRIPTION
**Pull request by shaniacht1**
fixes https://github.com/demisto/server/issues/7823

### CODE REVIEW

File: Integrations/integration-Endgame.yml [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+    };
+
+    var headerTransform = function(dict){
+        return function(obj){
+            return dict[obj];
+        };
+    };
+
+    //returns single object withing entity (i.e. File[0])
+    var jsonToEntityObject = function(origObj, keyTransform){
```
Comment (Spivaka) [inline Integrations/integration-Endgame.yml]: move this (with better naming) to common

File: Integrations/integration-Endgame.yml [hunk 2]
```diff
@@ -0,0 +1,783 @@
...
+                Headers: headers,
+                Body: data? JSON.stringify(data) : undefined
+            },
+            insecure,
+            proxy
+        );
+
+        if (res.StatusCode >= 300 || res.StatusCode < 200) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nResponse Body: ' + res.Body;
+        } else {
```
Comment (Spivaka) [inline Integrations/integration-Endgame.yml]: else is redundant

File: Integrations/integration-Endgame.yml [hunk 3]
```diff
@@ -0,0 +1,783 @@
...
+            try {
+                return JSON.parse(res.Body);
+            } catch (err) {
+                return res.Body;
```
Comment (Spivaka) [inline Integrations/integration-Endgame.yml]: log?

File: Integrations/integration-Endgame.yml [hunk 4]
```diff
@@ -0,0 +1,783 @@
...
+            results = deploy(args.domain, args.username, args.password, args.api_key, args.endpoints_transaction_id, args.deployment_profile, makeArray(args.endpoint_ips), makeArray(args.endpoint_ids));
+            if(results.submitted){
+                results.deployment_profile = args.deployment_profile;
+                results.id = results.submitted;
+                results.domain = args.domain;
+                results.submitted = undefined;
+            }
+            break;
+    }
+    logout();
```
Comment (Spivaka) [inline Integrations/integration-Endgame.yml]: what happen if the request fails? no logout?

Comment (shaniacht1) [reply Integrations/integration-Endgame.yml]: fixed

**Pull request merged**